src/ui/popupmenu/completion_item_widget.rs

```diff
@@ -20,14 +20,17 @@ pub struct CompletionItemWidgetWrap {
     /// Label displaying `menu` for this item in the list.
     pub menu: gtk::Label,
     /// Image of the item in the row.
-    pub kind: gtk::Image,
+    pub image: gtk::Image,
+    /// Kind of the item
+    pub kind: CompletionItemKind,
     /// Root container.
     pub row: gtk::ListBoxRow,
 }
 
 impl CompletionItemWidgetWrap {
     pub fn create(
         item: CompletionItem,
+        items: &Vec<CompletionItem>,
```

Comment:
Here, you should take parameter `show_kind` which is a boolean and tells if the kind image should be shown or not. Currently, either all the kind images are shown, or they are not.